exporters/otlp/all/src/main/java/io/opentelemetry/exporter/otlp/internal/OtlpConfigUtil.java

```diff
@@ -206,6 +211,44 @@ public static void configureOtlpHistogramDefaultAggregation(
     }
   }
 
+  /**
+   * Calls {@code #setMemoryMode} on the {@code Otlp{Protocol}MetricExporterBuilder} with the {@code
+   * memoryMode}.
+   */
+  public static void setMemoryModeOnOtlpMetricExporterBuilder(
+      Object builder, MemoryMode memoryMode) {
+    try {
+      if (builder instanceof OtlpGrpcMetricExporterBuilder) {
+        // Calling getDeclaredMethod causes all private methods to be read, which causes a
+        // ClassNotFoundException when running with the OkHttHttpProvider as the private
+        // setManagedChanel(io.grpc.ManagedChannel) is reached and io.grpc.ManagedChannel is not on
+        // the classpath. io.opentelemetry.exporter.otlp.metrics.OtlpGrpcMetricUtil provides a layer
+        // of indirection which avoids scanning the OtlpGrpcMetricExporterBuilder private methods.
+        Class<?> otlpGrpcMetricUtil =
+            Class.forName("io.opentelemetry.exporter.otlp.metrics.OtlpGrpcMetricUtil");
+        Method method =
+            otlpGrpcMetricUtil.getDeclaredMethod(
+                "setMemoryMode", OtlpGrpcMetricExporterBuilder.class, MemoryMode.class);
+        method.setAccessible(true);
+        method.invoke(null, builder, memoryMode);
+      } else if (builder instanceof OtlpHttpMetricExporterBuilder) {
+        Method method =
+            OtlpHttpMetricExporterBuilder.class.getDeclaredMethod(
+                "setMemoryMode", MemoryMode.class);
+        method.setAccessible(true);
+        method.invoke(builder, memoryMode);
```

Comment:
Wait why are we doing this with reflection instead of just casting and calling? Is it because we're trying very hard to keep it non-public or is there another reason I'm not immediately seeing.

Reply:
Is it for version interop?

Reply:
The `setMemoryMode` method is still package private and this code is in a different package. Just being conservative while I improve confidence about this config option. Goal is to make it public, maybe as soon as the next release.